core/modules/layout/layout.flexible.inc

```diff
@@ -577,20 +625,43 @@ function layout_flexible_template_edit_row_form($form, &$form_state, LayoutFlexi
   return $form;
 }
 
+/**
+ * Callback: Check if machine_name is unique.
+ *
+ * Even though callback is required for the Form API machine_name field type,
+ * the region classes are not required to be unique. It will accept any
+ * machine_name.
```

Comment:
I think not requiring regions to be unique may be a mistake. Region names are required to be unique in non-dynamic layouts (as defined in the layout.info file). We won't be able to make them non-unique in the future, so I think it's important that we specify this as required to be unique at this point.

Reply:
@quicksketch this one is tricky. We don't want to force region names to be unique across all flexible templates. Perhaps we only want to require them to be unique per row. But how do we pass the form info to the machine name's "exists" callback? I think we may need to have a hidden form field that only stores an array of the other regions in the row for the callback. And this field needs to be updated whenever one of the other machine names is updated - before it's saved.

Reply:
We'd only want to check on the current layout for sure, not across all layouts.

It looks like `exists` callbacks get passed 3 arguments: `$machine_name, $element, $form_state`

I'm looking at `entity_view_mode_exists()` as an example. So if you can pull the layout name from `$form_state` (or `$element` if necessary), you can use `layout_flexible_tempstore_load()` to check the _current_ values of the machine names of all regions.

I think something like this would work:
```
function _layout_flexible_template_region_class_name_exists($machine_name, $element, $form_state) {
  $flexible_template = layout_flexible_tempstore_load($form_state['flexible_template_name']);
  // Loops through the template regions to check for $machine_name...
}
```